pyvista/plotting/plotting.py

```diff
@@ -2325,10 +2327,17 @@ def add_mesh(
                     rgb = True
 
             original_scalar_name = scalars
-            scalars = get_array(mesh, scalars, preference=preference, err=True)
+            scalars, field = get_array(mesh, scalars, preference=preference, err=True)
             scalar_bar_args.setdefault('title', original_scalar_name)
             scalars_name = original_scalar_name
 
+            if field == FieldAssociation.POINT:
+                mesh.point_data.active_scalars_name = original_scalar_name
+                self.mapper.SetScalarModeToUsePointData()
+            elif field == FieldAssociation.CELL:
+                mesh.cell_data.active_scalars_name = original_scalar_name
+                self.mapper.SetScalarModeToUseCellData()
```

Comment:
Note to @adeak: you're going to say that this logic should be in `mapper.set_scalars`, and you're right. The mapper really needs to be refactored, but rather than doing this in a huge PR, I'd prefer doing it here in a hot-fix and leave the refactor to later.